src/node_api.cc

```diff
@@ -176,10 +178,8 @@ class ThreadSafeFunction : public node::AsyncResource {
         return napi_closing;
       }
     } else {
-      if (uv_async_send(&async) != 0) {
-        return napi_generic_failure;
-      }
       queue.push(data);
+      Send();
```

Comment:
I've swapped the order since it is treated as a hard failure and it is more logical to push before the notification.